app/src/instance-event/commons.js

```diff
@@ -0,0 +1,33 @@
+"use strict";
+
+const path = require("path");
+const _ = require("lodash");
+const S = require("string");
+
+const utils = require("../utils/utils");
+
+module.exports = {
+  /**
+   * Substitutes aliases found in an instance definition by their corresponding value.
+   *
+   * @param {Object} instanceDef - The Instance Definition on which to apply the substitution
+   *
+   * @return {Object} substitutedInstanceDef - The Instance Definition with its aliases substituted by their actual value
+   */
+  substituteAliases: function(instanceDef) {
+    if (!_.isEmpty(instanceDef)) {
```

Comment:
This check is a bit arbitrary IMO. I personally don't like big wrappers like that. I would rather check whether the instance is empty, and if yes, throw an error.

But again this is dragging us into validating an instance def. and I think it's out of scope here. I would just proceed directly with using the templating lib.

Reply:
This was just done to avoid NPE when later calling `instanceDef.uuid`.
So would the following be ok instead
```
    if (_.isEmpty(instanceDef)) {
      throw new Error("Illegal argument: empty instanceDef.");
    }
```
Or do we not even care doing the check at all?